engine/src/xcc.rs

```diff
@@ -0,0 +1,172 @@
+use aurora_engine_sdk::error::ReadU32Error;
+use aurora_engine_sdk::io::{StorageIntermediate, IO};
+use aurora_engine_sdk::promise::PromiseHandler;
+use aurora_engine_types::account_id::AccountId;
+use aurora_engine_types::parameters::{PromiseAction, PromiseBatchAction, PromiseCreateArgs};
+use aurora_engine_types::storage::{self, KeyPrefix};
+use aurora_engine_types::types::{Address, NearGas, Yocto, ZERO_YOCTO};
+use aurora_engine_types::Vec;
+use borsh::{BorshDeserialize, BorshSerialize};
+
+pub const ERR_NO_ROUTER_CODE: &str = "ERR_MISSING_XCC_BYTECODE";
+pub const ERR_CORRUPTED_STORAGE: &str = "ERR_CORRUPTED_XCC_STORAGE";
+pub const ERR_INVALID_ACCOUNT: &str = "ERR_INVALID_XCC_ACCOUNT";
+pub const VERSION_KEY: &[u8] = b"version";
+pub const CODE_KEY: &[u8] = b"router_code";
+pub const VERSION_UPDATE_GAS: NearGas = NearGas::new(5_000_000_000_000);
+pub const INITIALIZE_GAS: NearGas = NearGas::new(5_000_000_000_000);
+/// Amount of NEAR needed to cover storage for a router contract.
+pub const STORAGE_AMOUNT: Yocto = Yocto::new(5_000_000_000_000_000_000_000_000);
```

Comment:
2 NEAR is not enough for the tests to pass, so more storage staking must be needed. I reduced from 5 NEAR to 4 in https://github.com/aurora-is-near/aurora-engine/pull/560/commits/7815ee44ab96a51bde1e7cf578a09cbdbb4f423a

Reply:
This is probably because you are compiling the contract without optimizations applied. 

I use this line to compile the final binary:

`env RUSTFLAGS='-C link-arg=-s' cargo build --target wasm32-unknown-unknown --release`

Reply:
Yes, that did it, thanks! Fixed in https://github.com/aurora-is-near/aurora-engine/pull/560/commits/b9810cd9f49e8b40e6db4d2adec8f0e4b4518d5e